### ISSUE TITLE
Confusion about Fig.6

### ISSUE THREAD

**Issue by csustYyh** [hkchengrex/Mask-Propagation#44]
I apologize for bothering you.
Is Figure 6 generated by compiling relevant data from multiple video sequences, calculating their mean and interquartile range, and then plotting them? If so, could you please specify which dataset's video sequences you used?
![image](https://github.com/hkchengrex/Mask-Propagation/assets/118873316/e1f8776a-477e-4ad6-807f-6aa5e793772f)

**Comment by hkchengrex** [comment]
We used all sequences in the validation set. Some sequences are not as long so we drop them from the statistics computation once they reached their end frames.

**Comment by csustYyh** [comment]
Thank you for clearing up my confusion.
Best wishes!

**Comment by csustYyh** [comment]
Sorry for bothering again.
When calculating mIoU, is the probability map used or the results after applying argmax?
![image](https://github.com/hkchengrex/Mask-Propagation/assets/118873316/b4beae2d-ca52-457b-92dd-04dc2ea929b9)

**Comment by hkchengrex** [comment]
After. I am not sure why/how to do it before argmax.

**Comment by csustYyh** [comment]
I apologize for my shallow question.
Because I found that during the training process of STCN, probability maps are used to calculate IoU.
![image](https://github.com/hkchengrex/Mask-Propagation/assets/118873316/c34b88af-3833-4e9b-a4e1-9f439ea5155b)
![image](https://github.com/hkchengrex/Mask-Propagation/assets/118873316/b4f50a4c-d4f4-4403-8014-df74907561fd)
![image](https://github.com/hkchengrex/Mask-Propagation/assets/118873316/9e4ccbaa-be56-4c40-ab63-e3154932ca27)
I think that I confused training with inferencing.
Thanks for your kind anwser, I am a rookie in VOS, sometimes I appear a bit awkward.
If the calculation is based on the results after applying argmax, can I save the test results and then write a script to calculate the mIoU between the test results and ground truth?

**Comment by hkchengrex** [comment]
Don't worry about it :)
During training sometimes "soft IoU" might be used so that gradient can pass through. In the STCN code that you posted though, a threshold is used so "hard IoU" is computed anyway.
For actual evaluation, we almost always use "hard IoU", I believe.

> If the calculation is based on the results after applying argmax, can I save the test results and then write a script to calculate the mIoU between the test results and ground truth?

Yes.

**Comment by csustYyh** [comment]
Thank you very very very much for your guidance， I've gained a lot from it.
I have great admiration for you. Wishing you happiness every day! ^ ^
